### PR TITLE
cli watch option, unified logic for jobs and tjm

### PR DESCRIPTION
**Pull request by ciorg**

### CODE REVIEW

**Review by godber — CHANGES_REQUESTED** [review]

File: packages/teraslice-cli/src/cmds/jobs/restart.ts [hunk 1]
```diff
@@ -7,25 +7,32 @@ import reply from '../../helpers/reply';
...
+        yargs.options('interval', yargsOptions.buildOption('interval'));
+        yargs.options('watch', yargsOptions.buildOption('jobs-watch'));
         yargs.options('yes', yargsOptions.buildOption('yes'));
         yargs.strict()
-            .example('$0 jobs restart cluster1 99999999-9999-9999-9999-999999999999')
-            .example('$0 jobs restart cluster1 99999999-9999-9999-9999-999999999999 --yes')
-            .example('$0 jobs restart cluster1 --all');
+            .example('$0 jobs restart CLUSTER_ALIAS JOB_ID', 'restarts job on the cluster')
+            .example('$0 jobs restart CLUSTER_ALIAS JOB_ID1 JOB_ID2', 'restarts two jobs on the cluster')
+            .example('$0 jobs restart CLUSTER_ALIAS JOB_ID all --status failing', 'restarts all failing jobs on the cluster')
```
[the remark this answers is not in the document]
Comment (ciorg) [reply packages/teraslice-cli/src/cmds/jobs/restart.ts:24]: yeah - JOB_ID is a typo

File: packages/teraslice-cli/src/helpers/yargs-options.ts [hunk 2]
```diff
@@ -165,46 +170,57 @@ export default class Options {
...
             default: '_updated:desc'
         }),
+        'jobs-save': () => ({
+            describe: 'Saves controller and execution state a job or jobs locally after running command or uses a saved state file and applies commands to saved jobs',
+            default: false,
+            type: 'boolean'
+        }),
+        'jobs-watch': () => ({
+            alias: 'watch',
+            describe: 'Watches job for n number of slices, ensures correct number of workers and no failed slices.',
```
[the remark this answers is not in the document]
Comment (ciorg) [reply packages/teraslice-cli/src/helpers/yargs-options.ts:212]: It doesn't return a value, but exits with code 1 if the watch is not successful, ie failed slices or not enough workers after n slices.

There is a timeout which defaults to 2 minutes

File: packages/teraslice-cli/src/helpers/jobs.ts [hunk 3]
```diff
@@ -16,408 +29,697 @@ export default class Jobs {
...
+                }
+            } catch (e) {
+                reply.fatal(e);
+            }
         }
     }
 
-    async run(): Promise<void> {
-        await this.start();
+    async error() {
```
[the remark this answers is not in the document]
Comment (ciorg) [reply packages/teraslice-cli/src/helpers/jobs.ts]: yeah - there is definitely more refinement that could be done on this front.  I'm just not sure if I want to do it now.

File: packages/teraslice-cli/src/helpers/yargs-options.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (godber) [reply packages/teraslice-cli/src/helpers/yargs-options.ts:212]: Great!

**Review by godber — APPROVED** [review]

**Pull request merged**